pkg/operator/vspherecontroller/operator_removal.go

```diff
@@ -0,0 +1,135 @@
+		"rbac/lease_leader_election_rolebinding.yaml",
+		"controller_sa.yaml",
+		"controller_pdb.yaml",
+		"node_sa.yaml",
+		"csidriver.yaml",
+		"service.yaml",
+		"ca_configmap.yaml",
+		// Static assets used by the webhook
+		"webhook/config.yaml",
+		"webhook/sa.yaml",
+		"webhook/service.yaml",
+		"webhook/configuration.yaml",
+		"webhook/rbac/role.yaml",
+		"webhook/rbac/rolebinding.yaml",
+		"webhook/pdb.yaml",
+		// dynamic assets used by controller
+		"volumesnapshotclass.yaml",
+		"controller.yaml",
+		"node.yaml",
+		"servicemonitor.yaml",
+		"webhook/deployment.yaml",
+		"vsphere_cloud_config_secret.yaml",
+		"vsphere_features_config.yaml",
+	}
+
+	compositeClient := resourceapply.NewKubeClientHolder(c.kubeClient).WithDynamicClient(c.apiClients.DynamicClient)
+	applyResult := resourceapply.DeleteAll(ctx, compositeClient, c.eventRecorder, assets.ReadFile, assetsToBeDeleted...)
+	allErrors := []error{}
+	for _, result := range applyResult {
+		if result.Error != nil {
+			allErrors = append(allErrors, result.Error)
+		}
+	}
+	if len(allErrors) > 0 {
+		return utilerrors.NewAggregate(allErrors)
+	}
+
+	err := c.storageClassController.SyncRemove(ctx)
+	if err != nil {
+		return fmt.Errorf("error removing storageclass: %v", err)
+	}
+	return c.removeConditions(ctx, status)
+}
+
+func (c *VSphereController) getDisabledConditionName() string {
+	return c.name + "Disabled"
+}
+
+func (c *VSphereController) removeConditions(ctx context.Context, status *operatorapi.OperatorStatus) error {
+	updateFuncs := []v1helpers.UpdateStatusFunc{}
+	matchingConditions := []operatorapi.OperatorCondition{}
+
+	// make sure to not remove disabled condition
+	untouchableConditions.Insert(c.getDisabledConditionName())
+
+	originalConditions := status.DeepCopy().Conditions
+	for _, condition := range originalConditions {
+		if strings.HasPrefix(condition.Type, vsphereOperatorPrefix) && !untouchableConditions.Has(condition.Type) {
+			klog.Infof("Removing condition %s", condition.Type)
+			matchingConditions = append(matchingConditions, condition)
+		}
+	}
+	updateFuncs = append(updateFuncs, func(status *operatorapi.OperatorStatus) error {
+		for _, condition := range matchingConditions {
+			v1helpers.RemoveOperatorCondition(&status.Conditions, condition.Type)
+		}
+		return nil
+	})
+	// also add a condition to indicate that the operator is disabled
+	updateFuncs = append(updateFuncs, c.addDisabledCondition())
+	_, _, err := v1helpers.UpdateStatus(ctx, c.operatorClient, updateFuncs...)
+	return err
+}
+
+func (c *VSphereController) addDisabledCondition() v1helpers.UpdateStatusFunc {
+	return func(status *operatorapi.OperatorStatus) error {
+		disabledConditionName := c.getDisabledConditionName()
+		disabledCond := operatorapi.OperatorCondition{
+			Type:    disabledConditionName,
+			Status:  operatorapi.ConditionTrue,
+			Message: "Operator is disabled",
+		}
+		v1helpers.SetOperatorCondition(&status.Conditions, disabledCond)
+		return nil
```

Comment:
Usually this gets removed once - https://github.com/openshift/vmware-vsphere-csi-driver-operator/blob/master/pkg/operator/vspherecontroller/vspherecontroller.go#L579 code runs, but it appears to be buggy and sometimes this condition is not removed. 

I can reproduce this via e2e. :(